server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -91,7 +101,7 @@ public class IndexService extends AbstractIndexComponent implements IndicesClust
     private final ShardStoreDeleter shardStoreDeleter;
     private final QueryCache queryCache;
     private final IndexStorePlugin.DirectoryFactory directoryFactory;
-    private final Supplier<TableInfo> getTable;
+    private Supplier<TranslogIndexer> getTranslogIndexer;
```

Comment:
Maybe worth initializing this to something that raises a IllegalStateException or AssertionError indicating that it musn't be used before `validateMapping` is called